system/services/admin/DataManagerService.cfc

```diff
@@ -442,11 +442,19 @@ component {
 				args.filterParams[ param.name ] = args.filterParams[ param.name ] ?: param;
 			}
 		}
+		
+		var enhancedGridRecordcount = $isFeatureEnabled( "enhancedGridRecordcount" );
+		
+		if ( enhancedGridRecordcount ) {
+			args.selectFields.append( "count(*) over() as _total_recordcount" );
```

Comment:
is this ANSI SQL? compat with SQL server, MySQL and Postgres?

Reply:
I tested it in Mariadb and SQL Server. according to the inter web Postgres should work as well. It depends on the dbms engine version when the vendor introduced the functionality.
They are called window functions and Mariadb for instance supports count in window functions since version 10.2.

Reply:
Right kk, cool. I think if we're going to spend time at all on this, let's get it tested and working in the upcoming release branch. I'd suggest:

1. DB adapter method that signals support for the feature (defaults to false and can be dynamic - but the result cache for once in the lifetime of the app)
2. Ditch the feature flag
3. Add DB adapter method for getting the SQL to use (`"#sqlToUse# as _total_recordcount"`)

Reply:
I've just tested and can confirm that the syntax works in PostreSQL too...

Reply:
Unsupported engines (if there are any) could simply return NULL as the sqlToUse, and that would trigger a separate count query via the current method.

Reply:
Changes implemented as discussed. I decided to go with Dom's original idea and added an additional check method on the adapter in order to not have the extra select field added even if db does not support it.

Reply:
Nice - merging into upcoming release now.